pkg/operator/serviceaccountissuercontroller/serviceaccountissuer_controller.go

```diff
@@ -0,0 +1,206 @@
+		configInformer.Config().V1().Authentications().Informer(),
+	).ResyncEvery(30*time.Second).WithSync(ret.sync).ToController("ServiceAccountIssuerController", eventRecorder)
+}
+
+func (c *ServiceAccountIssuerController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	authConfig, err := c.authLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+	currentIssuer := authConfig.Spec.ServiceAccountIssuer
+
+	operator, err := c.kubeAPIserverOperatorLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	// there is no service account issuer set and there are no service account issuers in status, no-op.
+	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
+		return nil
+	}
+
+	// if the status is not set, but the current issuer is not empty, update the status and let the controller resync
+	if len(currentIssuer) > 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
+		operatorCopy := operator.DeepCopy()
+		operatorCopy.Status.ServiceAccountIssuers = []operatorv1.ServiceAccountIssuerStatus{
+			{
+				Name: currentIssuer,
+			},
+		}
+		_, statusUpdateErr := c.kubeAPIServerOperatorClient.UpdateStatus(ctx, operatorCopy, metav1.UpdateOptions{})
+		if statusUpdateErr == nil {
+			klog.Infof("ServiceAccountIssuer %q is now set as active issuer", currentIssuer)
+			statusUpdateErr = factory.SyntheticRequeueError
+		}
+		return statusUpdateErr
+	}
+
+	// in this case, user deleted service account issuer value, which means we have to remove all values from the status
+	// if we don't do this and instead just remove first value, the new tokens will be issued with "previous" service
+	// account issuer which is not desired
+	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) != 0 {
+		operatorCopy := operator.DeepCopy()
+		operatorCopy.Status.ServiceAccountIssuers = []operatorv1.ServiceAccountIssuerStatus{}
+		_, statusUpdateErr := c.kubeAPIServerOperatorClient.UpdateStatus(ctx, operatorCopy, metav1.UpdateOptions{})
+		if statusUpdateErr == nil {
+			klog.Infof("ServiceAccountIssuer value is empty, removing all previously used trusted issuers")
+			statusUpdateErr = factory.SyntheticRequeueError
+		}
+		return statusUpdateErr
+	}
+
+	activeIssuer := getActiveServiceAccountIssuer(operator.Status.ServiceAccountIssuers)
+	if len(activeIssuer) == 0 {
+		// at this point, we must have the active issuer (the one without expiration time).
+		// if we don't it means somebody changed the status deliberately.
+		// in this case, we correct it by setting the configured value as active.
+		// NOTE: this is an error/edge case
+		return c.makeActiveIssuerTrusted(ctx, currentIssuer, currentIssuer, operator)
+	}
+
+	// the issuer configured in auth config and the active issuer we have in operator status matches.
+	// this is no-op configuration wise, but we prune the list from expired issuers.
+	if currentIssuer == activeIssuer {
```

Comment:
stylistic nit for better readability:
```
issuerChanged := currentIssuer != activeIssuer

if issuerChanged {
  c.makeActiveIssuerTrusted(...)
} else {
  c.pruneExpiredServiceAccountIssuers(...)
}
```

Reply:
I don't like if-elsing that much... in my opinion, it makes the code read worse, and in addition, the comment blocks are nested.

Reply:
you have a point with else-structures. really just a nit, ignore at will